server/service/integration_core_test.go

```diff
@@ -2706,156 +2728,800 @@ func (s *integrationTestSuite) TestIntegrationsConfig() {
+	require.Equal(t, int64(123), config.Integrations.Zendesk[1].GroupID)
+	require.False(t, config.Integrations.Zendesk[1].EnableSoftwareVulnerabilities)
+
+	// delete first Zendesk integration
+	s.DoRaw("PATCH", "/api/v1/fleet/config", []byte(fmt.Sprintf(`{
+		"integrations": {
+			"zendesk": [
+				{
+					"url": %q,
+					"email": "test123@example.com",
+					"group_id": 123,
+					"enable_software_vulnerabilities": false
+				}
+			]
+		}
+	}`, srv.URL)), http.StatusOK)
+
+	config = s.getConfig()
+	require.Len(t, config.Integrations.Jira, 0)
+	require.Len(t, config.Integrations.Zendesk, 1)
+	require.Equal(t, int64(123), config.Integrations.Zendesk[0].GroupID)
+
+	// replace Zendesk integration
+	s.DoRaw("PATCH", "/api/v1/fleet/config", []byte(fmt.Sprintf(`{
+		"integrations": {
+			"zendesk": [
+				{
+					"url": %q,
+					"email": "ok@example.com",
+					"api_token": "ok",
+					"group_id": 122,
+					"enable_software_vulnerabilities": true
+				}
+			]
+		}
+	}`, srv.URL)), http.StatusOK)
+
+	config = s.getConfig()
+	require.Len(t, config.Integrations.Jira, 0)
+	require.Len(t, config.Integrations.Zendesk, 1)
+	require.Equal(t, int64(122), config.Integrations.Zendesk[0].GroupID)
+
+	// try adding Zendesk integration without sending API token
+	s.DoRaw("PATCH", "/api/v1/fleet/config", []byte(fmt.Sprintf(`{
+		"integrations": {
+			"zendesk": [
+				{
+					"url": %q,
+					"email": "ok@example.com",
+					"api_token": "ok",
+					"group_id": 122,
+					"enable_software_vulnerabilities": true
+				},
+				{
+					"url": %[1]q,
+					"email": "test123@example.com",
+					"group_id": 123,
+					"enable_software_vulnerabilities": false
+				}
+			]
+		}
+	}`, srv.URL)), http.StatusBadRequest)
+
+	// try adding Zendesk integration with masked API token
+	s.DoRaw("PATCH", "/api/v1/fleet/config", []byte(fmt.Sprintf(`{
+		"integrations": {
+			"zendesk": [
+				{
+					"url": %q,
+					"email": "ok@example.com",
+					"api_token": "ok",
+					"group_id": 122,
+					"enable_software_vulnerabilities": true
+				},
+				{
+					"url": %[1]q,
+					"email": "test123@example.com",
+					"api_token": "********",
+					"group_id": 123,
+					"enable_software_vulnerabilities": false
+				}
+			]
+		}
+	}`, srv.URL)), http.StatusBadRequest)
+
+	// edit Zendesk integration without sending API token
+	s.DoRaw("PATCH", "/api/v1/fleet/config", []byte(fmt.Sprintf(`{
+		"integrations": {
+			"zendesk": [
+				{
+					"url": %q,
+					"email": "ok@example.com",
+					"group_id": 122,
+					"enable_software_vulnerabilities": false
+				}
+			]
+		}
+	}`, srv.URL)), http.StatusOK)
+
+	config = s.getConfig()
+	require.Len(t, config.Integrations.Jira, 0)
+	require.Len(t, config.Integrations.Zendesk, 1)
+	require.Equal(t, int64(122), config.Integrations.Zendesk[0].GroupID)
+	require.False(t, config.Integrations.Zendesk[0].EnableSoftwareVulnerabilities)
+
+	// edit Zendesk integration with masked API token
```

Comment:
Same here, you could also try with explicit empty: `"api_token": "",` (should succeed IIRC).